### PR TITLE
refactoring the installers_controller responses

### PR DESCRIPTION
**Pull request by MaximilianMeister**
make it usable for crowbarctl

### CODE REVIEW

File: crowbar_framework/app/controllers/installers_controller.rb [hunk 1]
```diff
@@ -51,18 +51,38 @@ def start
       # the shell Process will be spawned in the background and therefore has
       # not a direct return value which we can use here
       if Crowbar::Installer.installing?
-        flash[:notice] = I18n.t(".installers.start.installation_ongoing")
+        respond_to do |format|
+          format.json do
+            head :im_used
```
Comment (MaximilianMeister) [inline crowbar_framework/app/controllers/installers_controller.rb]: just for reference: http://httpstatusdogs.com/226-im-used

not sure if there is some more suitable status-code... 226 refers to GET and we do a POST. any suggestions?

**Comment by MaximilianMeister** [comment]
previous gating was successful https://ci.suse.de/job/openstack-mkcloud/12092/

just added a `410` in case crowbar is already installed. so lets wait for the next one ...

File: crowbar_framework/app/controllers/installers_controller.rb [hunk 2]
```diff
@@ -42,27 +46,74 @@ def status
   # Provides the restful api call for
   # /installer/start 	POST 	triggers installation
   def start
```
Comment (tboerger) [inline crowbar_framework/app/controllers/installers_controller.rb]: A lot of code duplication, maybe you should create some variables and do the respond_to block only at the bottom of the method?

**Comment by tboerger** [comment]
Beside the code duplication :+1:

**Comment by MaximilianMeister** [comment]
> A lot of code duplication, maybe you should create some variables and do the respond_to block only at the bottom of the method?

@tboerger good idea. got rid of code duplication

**Comment by rsalevsky** [comment]
+1

**Pull request merged**